WordPress/src/main/java/org/wordpress/android/ui/media/MediaGalleryPickerActivity.java

```diff
@@ -97,24 +96,29 @@ protected void onCreate(Bundle savedInstanceState) {
         }
 
         setContentView(R.layout.media_gallery_picker_layout);
-        mGridView = (GridView) findViewById(R.id.media_gallery_picker_gridview);
-        mGridView.setMultiChoiceModeListener(this);
-        mGridView.setOnItemClickListener(this);
+        mRecycler = (RecyclerView) findViewById(R.id.recycler);
+
+        int numColumns = MediaGridAdapter.getColumnCount(this);
+        mGridManager = new GridLayoutManager(this, numColumns);
+        mRecycler.setLayoutManager(mGridManager);
+
         // TODO: We want to inject the image loader in this class instead of using a static field.
-        mGridAdapter = new MediaGridAdapter(this, mSite, null, 0, WordPress.sImageLoader);
-        mGridAdapter.setSelectedItems(selectedItems);
+        mGridAdapter = new MediaGridAdapter(this, mSite, WordPress.sImageLoader);
         mGridAdapter.setCallback(this);
```

Comment:
while not part of the PR in a direct way, I had to double check whether this could lead to potential leaks. I think we're safe, but asking you too to make sure: this would be a circular reference, right? so the Java GC would take care of this as the Activity wouldn't be chained in case it needs be destroyed. Can you confirm / double check please?